### PR TITLE
Q217, 219

### PR DESCRIPTION
**Pull request by VivianWenhuiLiang**

### CODE REVIEW

File: src/com/vivian/java/leetcode/Q217_ContainsDuplicate.java [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
  * Given an array of integers, find if the array contains any duplicates. Your
  * function should return true if any value appears at least twice in the array,
  * and it should return false if every element is distinct.
  */
 public class Q217_ContainsDuplicate {
+    public boolean cotainsDuplicate(int[] nums) {
+        if (nums.length <= 1) {
+            return false;
+        }
+        Set<Integer> set = new HashSet<>();
```
Comment (GeorgeZhiXu) [inline src/com/vivian/java/leetcode/Q217_ContainsDuplicate.java]: Good! Progress to interface, instead of implementation.

Comment (VivianWenhuiLiang) [inline src/com/vivian/java/leetcode/Q217_ContainsDuplicate.java]: yeah，I learn from the best teacher

Comment (GeorgeZhiXu) [inline src/com/vivian/java/leetcode/Q217_ContainsDuplicate.java]: Oh, aren't you sweet!

File: src/com/vivian/java/leetcode/Q219_ContainsDuplicateII.java [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
 public class Q219_ContainsDuplicateII {
-
+    public boolean containDuplicateII(int[] nums, int k) {
+        if (nums == null || nums.length <= 1 || k <= 0) {
+            return false;
+        }
+        Map<Integer, Integer> map = new HashMap<>();
+        for (int i = 0; i < nums.length; i++) {
+            if (map.containsKey(nums[i])) {
+                if (i - map.get(nums[i]) == k) {
```
Comment (GeorgeZhiXu) [inline src/com/vivian/java/leetcode/Q219_ContainsDuplicateII.java]: Why the condition here is ==?
By the problem statement, the absolute difference between i and j is at most k. 
Condition here seems to be <=.

File: src/com/vivian/java/leetcode/Q219_ContainsDuplicateII.java [hunk 3]
```diff
@@ -1,10 +1,28 @@
...
+                    return true;
+                }
+                return false;
```
Comment (GeorgeZhiXu) [inline src/com/vivian/java/leetcode/Q219_ContainsDuplicateII.java]: Return false here?

File: tst/com/vivian/java/leetcode/Q217_ContainsDuplicateTest.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.vivian.java.leetcode;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Unit test for Q28_strStr. In eclipse, you can right click this file, and run
```
Comment (GeorgeZhiXu) [inline tst/com/vivian/java/leetcode/Q217_ContainsDuplicateTest.java]: The comments are wrong!
Remove them.

File: tst/com/vivian/java/leetcode/Q219_ContainsDuplicateIITest.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.vivian.java.leetcode;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Unit test for Q28_strStr. In eclipse, you can right click this file, and run
```
Comment (GeorgeZhiXu) [inline tst/com/vivian/java/leetcode/Q219_ContainsDuplicateIITest.java]: The comments are wrong, remove them.

File: tst/com/vivian/java/leetcode/Q219_ContainsDuplicateIITest.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ * as JUnit Test.
+ */
+public class Q219_ContainsDuplicateIITest {
+    private Q219_ContainsDuplicateII solution = new Q219_ContainsDuplicateII();
+
+    @Test
+    public void test() {
```
Comment (GeorgeZhiXu) [inline tst/com/vivian/java/leetcode/Q219_ContainsDuplicateIITest.java]: This test doesn't seem to have covered all cases.

**Pull request merged**